File: apps/core-desktop/src/devData/sales/sales.ts
import type { Sale, SaleLine } from '@40labs/types';
import { WORKSPACE_ID, BRANCH_ID, daysAgoIso, nowIso } from '../constants';

const mockSaleLines1: SaleLine[] = [
  {
    id: 'saleline_001',
    inventory_item_id: 'inv_001',
    medicine_id: 'med_001',
    quantity: 2,
    unit_price: 3000,
    subtotal: 6000,
    dispensed_by_user_id: 'user_002',
    is_prescription_dispense: false,
  },
];

const mockSaleLines2: SaleLine[] = [
  {
    id: 'saleline_002',
    inventory_item_id: 'inv_002',
    medicine_id: 'med_002',
    quantity: 1,
    unit_price: 5000,
    subtotal: 5000,
    dispensed_by_user_id: 'user_003',
    is_prescription_dispense: true,
  },
];

export const initialSales: Sale[] = [
  {
    id: 'sale_001',
    workspace_id: WORKSPACE_ID,
    branch_id: BRANCH_ID,
    created_at: daysAgoIso(1),
    updated_at: daysAgoIso(1),
    customer_id: 'cust_001',
    lines: mockSaleLines1,
    payment_method: 'cash',
    discount_amount: 0,
    discount_authorized_by_user_id: null,
    tax_amount: 0,
    grand_total: 6000,
    currency: 'TZS',
    synced_at: daysAgoIso(1),
  },
  {
    id: 'sale_002',
    workspace_id: WORKSPACE_ID,
    branch_id: BRANCH_ID,
    created_at: nowIso(),
    updated_at: nowIso(),
    customer_id: null,
    lines: mockSaleLines2,
    payment_method: 'mobile_money',
    discount_amount: 0,
    discount_authorized_by_user_id: null,
    tax_amount: 0,
    grand_total: 5000,
    currency: 'TZS',
    synced_at: null,
  },
];

File: apps/core-desktop/src-tauri/src/models/sales.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct SaleLine {
    pub id: String,
    pub sale_id: String,
    pub inventory_item_id: String,
    pub medicine_name: String,
    pub quantity: i64,
    pub unit_price: i64,
    pub subtotal: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Sale {
    pub id: String,
    pub workspace_id: String,
    pub branch_id: String,
    pub created_at: String,
    pub updated_at: String,
    pub customer_id: Option<String>,
    pub lines: Vec<SaleLine>,
    pub payment_method: String,
    pub discount_amount: i64,
    pub discount_authorized_by_user_id: Option<String>,
    pub tax_amount: i64,
    pub grand_total: i64,
    pub currency: String,
    pub synced_at: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateSaleLineInput {
    pub inventory_item_id: String,
    pub medicine_name: String,
    pub quantity: i64,
    pub unit_price: i64,
    pub subtotal: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateSaleRequest {
    pub customer_id: Option<String>,
    pub lines: Vec<CreateSaleLineInput>,
    pub payment_method: String,
    pub discount_amount: i64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct FiscalReceipt {
    pub id: String,
    pub sale_id: String,
    pub receipt_number: String,
    pub qr_code_url: Option<String>,
    pub verification_status: String,
    pub created_at: String,
}
